ms-engine-spring-jaxrs/src/main/java/net/trajano/ms/engine/jaxrs/JaxRsPath.java

```diff
@@ -0,0 +1,211 @@
+
+    private final boolean post;
+
+    private final String[] produces;
+
+    private final boolean put;
+
+    public JaxRsPath(final String path,
+        final String[] consumes,
+        final String[] produces,
+        final boolean get,
+        final boolean post,
+        final boolean put,
+        final boolean delete) {
+
+        this.path = path;
+        this.consumes = consumes;
+        this.produces = produces;
+        this.get = get;
+        this.post = post;
+        this.put = put;
+        this.delete = delete;
+        final Pattern placeholderPattern = Pattern.compile("/\\{([^}]+)}");
+        final Pattern regexPlaceholderPattern = Pattern.compile("[-A-Za-z_0-9]+:\\s*(.+)");
+        final Matcher matcher = placeholderPattern.matcher(path);
+
+        final StringBuffer b = new StringBuffer();
+        while (matcher.find()) {
+            final Matcher m2 = regexPlaceholderPattern.matcher(matcher.group(1));
+            if (m2.matches()) {
+                matcher.appendReplacement(b, "/" + m2.group(1));
+            } else {
+                matcher.appendReplacement(b, "/[^/]+");
+            }
+        }
+        matcher.appendTail(b);
+        pathRegex = b.toString();
+        exact = pathRegex.equals(path);
+    }
+
+    /**
+     * Compares two JaxRsPath objects such that it is ordered by most specific and
+     * lowest level first. It sorts it in reverse by path. {@inheritDoc}
+     */
+    @Override
+    public int compareTo(final JaxRsPath o) {
+
+        if (exact && !o.exact) {
+            return -1;
+        } else if (!exact && o.exact) {
+            return 1;
+        }
+
+        return o.path.compareTo(path);
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final JaxRsPath other = (JaxRsPath) obj;
+        if (!Arrays.equals(consumes, other.consumes)) {
+            return false;
+        }
+        if (delete != other.delete) {
+            return false;
+        }
+        if (exact != other.exact) {
+            return false;
+        }
+        if (get != other.get) {
+            return false;
+        }
+        if (path == null) {
+            if (other.path != null) {
+                return false;
+            }
+        } else if (!path.equals(other.path)) {
+            return false;
+        }
+        if (pathRegex == null) {
+            if (other.pathRegex != null) {
+                return false;
+            }
+        } else if (!pathRegex.equals(other.pathRegex)) {
+            return false;
+        }
+        if (post != other.post) {
+            return false;
+        }
+        if (!Arrays.equals(produces, other.produces)) {
+            return false;
+        }
+        if (put != other.put) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)